src/langchain_google_alloydb_pg/utils/pgvector_migrator.py

```diff
@@ -0,0 +1,332 @@
+        delete_pg_collection (bool): An option to delete the original data upon migration.
+            Default: False. Optional.
+        insert_batch_size (int): Number of rows to insert at once in the table.
+            Default: 1000.
+    """
+    destination_table = vector_store.get_table_name()
+
+    # Get row count in PGVector collection
+    uuid_task = asyncio.create_task(_aget_collection_uuid(engine, collection_name))
+    query = (
+        f"SELECT COUNT(*) FROM {EMBEDDINGS_TABLE} WHERE collection_id=:collection_id"
+    )
+    async with engine._pool.connect() as conn:
+        uuid = await uuid_task
+        result = await conn.execute(text(query), parameters={"collection_id": uuid})
+        result_map = result.mappings()
+        collection_data_len = result_map.fetchone()
+    if collection_data_len is None:
+        warnings.warn(f"Collection, {collection_name} contains no elements.")
+        return
+
+    # Extract data from the collection and batch insert into the new table
+    data_batches = _aextract_pgvector_collection(
+        engine, collection_name, batch_size=insert_batch_size
+    )
+    await _concurrent_batch_insert(data_batches, vector_store, max_concurrency=100)
+
+    # Validate data migration
+    query = f"SELECT COUNT(*) FROM {destination_table}"
+    async with engine._pool.connect() as conn:
+        result = await conn.execute(text(query))
+        result_map = result.mappings()
+        table_size = result_map.fetchone()
+    if not table_size:
+        raise ValueError(f"Table: {destination_table} does not exist.")
+
+    if collection_data_len["count"] != table_size["count"]:
+        raise ValueError(
+            "All data not yet migrated.\n"
+            f"Original row count: {collection_data_len['count']}\n"
+            f"Collection table, {destination_table} row count: {table_size['count']}"
+        )
+    elif delete_pg_collection:
+        # Delete PGVector data
+        query = f"DELETE FROM {EMBEDDINGS_TABLE} WHERE collection_id=:collection_id"
+        async with engine._pool.connect() as conn:
+            await conn.execute(text(query), parameters={"collection_id": uuid})
+            await conn.commit()
+
+        query = f"DELETE FROM {COLLECTIONS_TABLE} WHERE name=:collection_name"
+        async with engine._pool.connect() as conn:
+            await conn.execute(
+                text(query), parameters={"collection_name": collection_name}
+            )
+            await conn.commit()
+        print(f"Successfully deleted PGVector collection, {collection_name}")
+
+
+async def _alist_pgvector_collection_names(
+    engine: AlloyDBEngine,
+) -> List[str]:
+    """Lists all collection names present in PGVector table."""
+    try:
+        query = f"SELECT name from {COLLECTIONS_TABLE}"
+        async with engine._pool.connect() as conn:
+            result = await conn.execute(text(query))
+            result_map = result.mappings()
+            all_rows = result_map.fetchall()
```

Comment:
nit: can we call "result.fetchall()" directly? https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.fetchmany